vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -200,18 +196,7 @@ def pre_build(self):
                     break
 
         # Set appropriate properties for the selector.
-        if isinstance(self.selector, SELECTORS["numerical"] + SELECTORS["temporal"]):
-            self.selector = cast(NumericalTemporalSelectorType, self.selector)
-            _min, _max = self._get_min_max(targeted_data)
-            # Note that manually set self.selector.min/max = 0 are Falsey but should not be overwritten.
-            if self.selector.min is None:
-                self.selector.min = _min
-            if self.selector.max is None:
-                self.selector.max = _max
-        else:
-            # Categorical selector.
-            self.selector = cast(CategoricalSelectorType, self.selector)
-            self.selector.options = self.selector.options or self._get_options(targeted_data)
+        self._set_selector_properties(targeted_data=targeted_data)
```

Comment:
Please can we put this back inline rather than a new function? I know that the code in `pre_build` is pretty long but overall I'd prefer it that way since we're likely to come back and change this stuff again in future, and splitting it out will make that a bit harder.